staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go

```diff
@@ -45,6 +45,18 @@ var PanicHandlers = []func(interface{}){logPanic}
 // E.g., you can provide one or more additional handlers for something like shutting down go routines gracefully.
 func HandleCrash(additionalHandlers ...func(interface{})) {
 	if r := recover(); r != nil {
+		HandleRecover(r, additionalHandlers...)
+	}
+}
+
+// HandleRecover takes a recover interface{} and logs an error. Meant to be called via
+// defer in cases where custom recover handling is needed.  Additional context-specific handlers can be provided, and will be
+// called in case of panic.  HandleCrash actually crashes, after calling the
```

Comment:
doc is weird/wrong... some copy/paste issues (`HandleCrash actually crashes`), some confusing bits (`takes a recover interface{}`)

Reply:
it's a bit unclear whether a caller should choose HandleRecover or HandleCrash

Reply:
rewrote to use the additional handler which does what I want.